### PR TITLE
#1280 step 1

### PR DESCRIPTION
**Pull request by youennmerel**
- implement a new export by examination id in dataset microservice
- generate input.json as specified
- integrate input.json to the new examination export when format is dicom

### CODE REVIEW

File: : [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# .bashrc
```
Comment (michaelkain) [inline ::1]: Dear Youenn, thx for your PR! Not sure, if this file should be part of it? Thank you and with kind regards, Michael

File: shanoir-ng-datasets/src/main/java/org/shanoir/ng/utils/DatasetFileUtils.java [hunk 2]
```diff
@@ -50,10 +51,13 @@ public static File getUserImportDir(String importDir) {
     public static void getDatasetFilePathURLs(final Dataset dataset, final List<URL> pathURLs, final DatasetExpressionFormat format) throws MalformedURLException {
         List<DatasetExpression> datasetExpressions = dataset.getDatasetExpressions();
         for (Iterator<DatasetExpression> itExpressions = datasetExpressions.iterator(); itExpressions.hasNext();) {
+            System.out.println("TEST - HAS DatasetExpression");
```
Comment (michaelkain) [inline shanoir-ng-datasets/src/main/java/org/shanoir/ng/utils/DatasetFileUtils.java:54]: Dear Youenn, please remove System.out.println, thank you!

File: shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/service/DatasetServiceImpl.java [hunk 3]
```diff
@@ -34,6 +34,7 @@
 import org.shanoir.ng.dataset.repository.DatasetRepository;
 import org.shanoir.ng.datasetfile.DatasetFile;
 import org.shanoir.ng.dicom.web.service.DICOMWebService;
+import org.shanoir.ng.dataset.dto.InputDTO;
```
Comment (michaelkain) [inline shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/service/DatasetServiceImpl.java:37]: not sure, if this import is need?

File: : (hunk 1 above)
Comment (youennmerel) [reply :]: Wow, no idea how this happened !

**Comment by youennmerel** [comment]
Hi Michael,

Thank you for your review !

I've now taken your comments into account, corrections commit has been squashed into https://github.com/fli-iam/shanoir-ng/pull/1571/commits/608bdfb9aa7306ec64dc9bf954ce64f41c134099.

**Comment by michaelkain** [comment]
Dear Youenn, thank you! First of all, pretty impressive PR, for your first PR!
I think, it could be great to remove as well the file examination.component.ts from your PR, as it does not really contain changes and the PR is a backend PR.

**Comment by michaelkain** [comment]
Dear Jean-Come @jcomedouteau,
I hope you are doing fine. I have added some quite trivial comments to the PR yesterday.
As you are our download specialist and currently working on CARMIN integration optimizations (multi-dataset processing) in the front, it would be a great help, if you could review the PR please. I will review the QC PR of Julien. I hope this is fine? and with kind regards, Michael PS: I will mutually ask Youenn to have a look on your PR, that you both are aware of your devs.

File: shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/repository/DatasetRepository.java [hunk 4]
```diff
@@ -38,4 +39,7 @@ public interface DatasetRepository extends PagingAndSortingRepository<Dataset, L
 
 	void deleteByIdIn(List<Long> ids);
 
+	@Query("SELECT d FROM Dataset d INNER JOIN DatasetAcquisition a WHERE a.examination.id = :examId")
+	Iterable<Dataset> findByExaminationId(Long examId);
```
Comment (jcomedouteau) [inline shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/repository/DatasetRepository.java:43]: I think you can directly do a findByDatasetAcquisitionExaminationId or findByDatasetAcquisition_ExaminationId without @Query annotation.
See https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#repositories.query-methods.query-property-expressions

**Comment by youennmerel** [comment]
@michaelkain After discussion with @jcomedouteau, I will finally add the front end modification to this PR, as it will greatly simplify the testing of the feature.

**Comment by youennmerel** [comment]
@jcomedouteau Please see my new commit !

**Review by jcomedouteau — APPROVED** [review]

**Pull request merged**